program/src/processor/initialize_stake.rs

```diff
@@ -10,12 +18,101 @@ use crate::instruction::accounts::{Context, InitializeStakeAccounts};
 ///
 /// 0. `[]` Stake config account
 /// 1. `[w]` Validator stake account
-///     * PDA seeds: ['stake', validator, config_account]
+///     * PDA seeds: ['stake::state::stake', validator, config]
 /// 2. `[]` Validator vote account
 /// 3. `[]` System program
 pub fn process_initialize_stake(
-    _program_id: &Pubkey,
-    _ctx: Context<InitializeStakeAccounts>,
+    program_id: &Pubkey,
+    ctx: Context<InitializeStakeAccounts>,
 ) -> ProgramResult {
+    // Accounts validation.
+
+    // config
+    // - owner must the stake program
+    // - must be initialized
+
+    require!(
+        ctx.accounts.config.owner == program_id,
+        ProgramError::InvalidAccountOwner,
+        "config"
+    );
+
+    let data = &ctx.accounts.config.try_borrow_data()?;
+
+    let config = bytemuck::try_from_bytes::<Config>(data)
+        .map_err(|_error| ProgramError::InvalidAccountData)?;
+
+    require!(
+        config.is_initialized(),
+        ProgramError::UninitializedAccount,
+        "config"
+    );
+
+    // validator_vote
+    // - owner must be the vote program
+    // - must be initialized
+
+    require!(
+        ctx.accounts.validator_vote.owner == &solana_program::vote::program::ID,
+        ProgramError::InvalidAccountOwner,
+        "validator_vote"
+    );
+
+    let data = &ctx.accounts.validator_vote.try_borrow_data()?;
+
+    require!(
+        VoteState::is_correct_size_and_initialized(data),
+        ProgramError::InvalidAccountData,
+        "validator_vote"
+    );
+
+    let validator = Pubkey::from(*array_ref!(data, 4, 32));
```

Comment:
Sorry that this wasn't clear -- let's use the validator vote address instead of the node id, since the node id can be reused in multiple vote accounts, which could easily let someone impersonate another node

Reply:
Created a PR for it: #30